mars/metrics/api.py

```diff
@@ -35,25 +36,39 @@
 
 
 def init_metrics(backend="console", config: Dict[str, Any] = None):
-    backend = backend or "console"
+    global _init
+    if _init is True:
+        return
     if backend not in _backends_cls:
         raise NotImplementedError(f"Do not support metric backend {backend}")
     global _metric_backend
     _metric_backend = backend
     if _metric_backend == "prometheus":
         try:
             from prometheus_client import start_http_server
+            from ..utils import get_next_port
 
             port = config.get("port", 0) if config else 0
+            port = port or get_next_port()
             start_http_server(port)
-            logger.info("Finished startup prometheus http server and port is %d", port)
+            logger.warning(
+                "Finished startup prometheus http server and port is %d", port
+            )
         except ImportError:
             logger.warning(
                 "Failed to start prometheus http server because there is no prometheus_client"
             )
+    _init = True
     logger.info("Finished initialize the metrics with backend %s", _metric_backend)
 
 
+def shutdown_metrics():
+    global _metric_backend
+    _metric_backend = "console"
+    global _init
+    _init = False
```

Comment:
The `shutdown_metrics` only set the global var `_init` to `False`, how does it shutdown the http server started by the prometheus metrics backend?

Reply:
Thanks for you reminder. We should do some some cleanup. But there is no `stop` or `shutdown` in `prometheus_client`.